src/main/java/org/apache/flink/benchmark/ContinuousFileReaderOperatorBenchmark.java

```diff
@@ -0,0 +1,118 @@
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.sink.DiscardingSink;
+import org.apache.flink.streaming.api.functions.source.ContinuousFileReaderOperator;
+import org.apache.flink.streaming.api.functions.source.SourceFunction;
+import org.apache.flink.streaming.api.functions.source.TimestampedFileInputSplit;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.OperationsPerInvocation;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+@OperationsPerInvocation(value = ContinuousFileReaderOperatorBenchmark.RECORDS_PER_INVOCATION)
+public class ContinuousFileReaderOperatorBenchmark extends BenchmarkBase {
+    private static final int SPLITS_PER_INVOCATION = 100;
+    private static final int LINES_PER_SPLIT = 20_000_000;
+    public static final int RECORDS_PER_INVOCATION = SPLITS_PER_INVOCATION * LINES_PER_SPLIT;
+
+    private static final TimestampedFileInputSplit SPLIT = new TimestampedFileInputSplit(0, 0, new Path("."), 0, 0, new String[]{});
+    private static final String LINE = Strings.repeat('0', 10);
+
+    public static void main(String[] args)
+            throws RunnerException {
+        Options options = new OptionsBuilder()
+                .verbosity(VerboseMode.NORMAL)
+                .include(".*" + ContinuousFileReaderOperatorBenchmark.class.getSimpleName() + ".*")
+                .build();
+
+        new Runner(options).run();
+    }
+
+    @Benchmark
+    public void readFiles(FlinkEnvironmentContext context) throws Exception {
+        StreamExecutionEnvironment env = context.env;
+        env
+                .enableCheckpointing(100)
+                .setParallelism(1)
+                .addSource(buildSourceFun())
+                .transform("fileReader", TypeInformation.of(String.class),
+                        new ContinuousFileReaderOperator<>(new MockInputFormat()))
+                .addSink(new DiscardingSink<>());
+
+        env.execute();
+    }
+
+    private SourceFunction<TimestampedFileInputSplit> buildSourceFun() {
+        return new SourceFunction<TimestampedFileInputSplit>() {
+            private volatile boolean isRunning = true;
+            private int count = 0;
+
+            @Override
+            public void run(SourceContext<TimestampedFileInputSplit> ctx) {
+                while (isRunning && count < SPLITS_PER_INVOCATION) {
+                    count++;
+                    synchronized (ctx.getCheckpointLock()) {
+                        ctx.collect(SPLIT);
+                    }
+                }
+            }
+
+            @Override
+            public void cancel() {
+                isRunning = false;
+            }
+        };
+    }
```

Comment:
convert into a named `private static` class ? Calling `buildSourceFun()` is a bit strange equivalent of calling a constructor.